pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import itertools as it
+import re
+
+from pyfr.inifile import Inifile
+from pyfr.integrators.dual.pseudo.base import BaseDualPseudoIntegrator
+from pyfr.util import memoize, proxylist
+
+
+class DualMultiPIntegrator(BaseDualPseudoIntegrator):
+    def __init__(self, backend, systemcls, rallocs, mesh, initsoln, cfg,
+                 tcoeffs, multip_types):
+        self.backend = backend
+
+        sect = 'solver-dual-time-integrator-multip'
+        # Get the solver order
+        self._order = cfg.getint('solver', 'order')
+
+        # Get the multigrid cycle
+        self.cycle, self.csteps = zip(*cfg.getliteral(sect, 'cycle'))
+        self.levels = sorted(set(self.cycle), reverse=True)
+        self.level = self._order
+
+        if max(self.cycle) > self._order:
+            raise ValueError('The multigrid level orders cannot exceed '
+                             'the solution order')
+
+        if any(abs(i - j) > 1 for i, j in zip(self.cycle, self.cycle[1:])):
+            raise ValueError('The orders of consecutive multigrid levels can '
+                             'only change by one')
+
+        if self.cycle[0] != self._order or self.cycle[-1] != self._order:
+            raise ValueError('The multigrid cycle needs to start end with the '
+                             'highest (solution) order ')
+
+        # Multigrid pseudo-time steps
+        dtau = cfg.getfloat('solver-time-integrator', 'pseudo-dt')
+        dtauf = cfg.getfloat(sect, 'pseudo-dt-fact', 1.0)
+        self.dtaus = {l: dtau*dtauf**(self._order - l) for l in self.levels}
+
+        self._maxniters = cfg.getint('solver-time-integrator',
```

Comment:
Declare a variable = 'solver-time-integrator' as we do above with sect.  Avoids a lot of linebreaks.